src/lib/field-next/field.ts

```diff
@@ -0,0 +1,203 @@
+  popoverIcon: boolean;
+  popoverExpanded: boolean;
+  multiline: boolean;
+  supportTextInset: FieldSupportTextInset;
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'forge-field': IFieldComponent;
+  }
+}
+
+/**
+ * @tag forge-field
+ * 
+ * @summary The Forge Field component is a basic component that handles the layout and theming of
+ * form elements that can include a label, various states, and a border around an input area.
+ * 
+ * @property {FieldLabelPosition} labelPosition - The position of the label relative to the input area.
+ * @property {FieldLabelAlignment} labelAlignment - The alignment of the label relative to the input area.
+ * @property {boolean} invalid - Whether the field is in an invalid state.
+ * @property {boolean} required - Whether the field is required.
+ * @property {boolean} optional - Whether the field is optional.
+ * @property {boolean} disabled - Whether the field is disabled.
+ * @property {FieldVariant} variant - The variant of the field.
+ * @property {FieldTheme} theme - The theme of the field.
+ * @property {FieldDensity} density - The density of the field.
+ * @property {boolean} dense - Whether the field is at the "small" density level.
+ * 
+ * @attribute {FieldLabelPosition} label-position - The position of the label relative to the input area.
+ * @attribute {FieldLabelAlignment} label-alignment - The alignment of the label relative to the input area.
+ * @attribute {boolean} invalid - Whether the field is in an invalid state.
+ * @attribute {boolean} required - Whether the field is required.
+ * @attribute {boolean} optional - Whether the field is optional.
+ * @attribute {boolean} disabled - Whether the field is disabled.
+ * @attribute {FieldVariant} variant - The variant of the field.
+ * @attribute {FieldTheme} theme - The theme of the field.
+ * @attribute {Density} density - The density of the field.
+ * @attribute {boolean} dense - Whether the field is at the "small" density level.
+ */
+
+@CustomElement({
+  name: FIELD_CONSTANTS.elementName,
+  dependencies: [FocusIndicatorComponent]
+})
+export class FieldComponent extends BaseComponent implements IFieldComponent {
+  public static get observedAttributes(): string[] {
+    return [
+      FIELD_CONSTANTS.attributes.LABEL_POSITION,
+      FIELD_CONSTANTS.attributes.LABEL_ALIGNMENT,
+      FIELD_CONSTANTS.attributes.FLOAT_LABEL,
+      FIELD_CONSTANTS.attributes.INVALID,
+      FIELD_CONSTANTS.attributes.REQUIRED,
+      FIELD_CONSTANTS.attributes.OPTIONAL,
+      FIELD_CONSTANTS.attributes.DISABLED,
+      FIELD_CONSTANTS.attributes.VARIANT,
+      FIELD_CONSTANTS.attributes.THEME,
+      FIELD_CONSTANTS.attributes.SHAPE,
+      FIELD_CONSTANTS.attributes.DENSITY,
+      FIELD_CONSTANTS.attributes.DENSE,
+      FIELD_CONSTANTS.attributes.POPOVER_ICON,
+      FIELD_CONSTANTS.attributes.POPOVER_EXPANDED,
+      FIELD_CONSTANTS.attributes.MULTILINE,
+      FIELD_CONSTANTS.attributes.SUPPORT_TEXT_INSET
+    ];
```

Comment:
Let's use our new `observedAttributes` pattern here so new observed attributes can be added automatically.